File: ai/src/main/java/org/pingel/causality/RandomBooleanFunction.java

package org.pingel.causality;

import org.pingel.bayes.Case;
import org.pingel.bayes.RandomVariable;
import org.pingel.bayes.Value;
import org.pingel.type.Booleans;

public class RandomBooleanFunction extends Function
{
	private double p;

	public RandomBooleanFunction(RandomVariable var, double p)
	{
		super(var);
		this.p = p;
	}
	
	protected Value compute(CausalModel m, Case memo)
	{
	    return (Math.random() < p) ? 
	            Booleans.tVal :
	                Booleans.fVal;
	}

}

File: ai/src/main/scala/org/pingel/causality/RandomBooleanFunction.scala

package org.pingel.causality

import org.pingel.bayes.Case
import org.pingel.bayes.RandomVariable
import org.pingel.bayes.Value
import org.pingel.ptype.PBooleansValues

class RandomBooleanFunction(rv: RandomVariable, p: Double) extends Function(rv, Nil) {

	def compute(m: CausalModel, memo: Case) = (scala.math.random < p) match {
		case true => PBooleansValues.tVal
		case false => PBooleansValues.fVal
	}

}
